libpod/container_attach.go

```diff
@@ -0,0 +1,144 @@
+package libpod
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/docker/docker/pkg/term"
+	"github.com/kubernetes-incubator/cri-o/utils"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+	"k8s.io/client-go/tools/remotecommand"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"net"
+)
+
+/* Sync with stdpipe_t in conmon.c */
+const (
+	AttachPipeStdin  = 1
+	AttachPipeStdout = 2
+	AttachPipeStderr = 3
+)
+
+// attachContainerSocket connects to the container's attach socket and deals with the IO
+func (c *Container) attachContainerSocket(resize <-chan remotecommand.TerminalSize, noStdIn bool, detachKeys []byte) error {
+	inputStream := os.Stdin
+	outputStream := os.Stdout
+	errorStream := os.Stderr
+	defer inputStream.Close()
+	tty, err := strconv.ParseBool(c.runningSpec.Annotations["io.kubernetes.cri-o.TTY"])
```

Comment:
Does this need to be an annotation check? We have a Terminal bool in the spec itself that doesn't require us to parse the annotations.

Reply:
I added this because at one time, it needed to be set for tty's to work.  I think we can probably deprecate it.

Reply:
I think we should just check c.runningSpec.Process.Terminal instead - it's already a bool

Reply:
Ok, but it is standard crio annotation.  Do you want it removed from there as well ?

Reply:
No need to remove the annotation, we just don't need to check against it.